hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java

```diff
@@ -105,7 +105,9 @@ public void handle(@NonNull final HandleContext handleContext) {
                 .grpcCertificateHash(op.grpcCertificateHash())
                 .declineReward(op.declineReward())
                 .adminKey(op.adminKey());
-        final var node = nodeBuilder.nodeId(getNextNodeID(nodeStore)).build();
+        // Since nodes won't be removed from state, we can set the nodeId to the next available id
+        // in the state based on the size of the state.
+        final var node = nodeBuilder.nodeId(nodeStore.sizeOfState()).build();
```

Comment:
I guess `sizeOfState()` counts the number of entities that have been deleted too, otherwise this wouldn't work, yes?